File: java/src/main/java/customer/Store.java
package customer;

import io.vavr.collection.Map;
import lombok.AllArgsConstructor;

@AllArgsConstructor
public class Store {
    private final Map<ProductType, Integer> inventory;

    public Store removeInventory(ProductType product, Integer quantity) {
        return updateStore(product, getInventoryFor(product) - quantity);
    }

    public Store addInventory(ProductType product, Integer quantity) {
        return updateStore(product, getInventoryFor(product) + quantity);
    }

    public boolean hasEnoughInventory(ProductType product, Integer quantity) {
        return getInventoryFor(product) >= quantity;
    }

    public Integer getInventoryFor(ProductType product) {
        return inventory.getOrElse(product, 0);
    }

    private Store updateStore(ProductType product, Integer newQuantity) {
        return new Store(inventory.put(product, newQuantity));
    }
}

File: php/src/Customer/Store.php
<?php

namespace Com\CleanTests\Customer;

class Store
{
    private array $inventory;

    public function __construct()
    {
        $this->inventory = [];
    }

    public function addInventory(ProductType $productType, int $quantity): void
    {
        $this->updateStore($productType, $this->getInventoryFor($productType) + $quantity);
    }

    public function removeInventory(ProductType $productType, int $quantity): void
    {
        $this->updateStore($productType, $this->getInventoryFor($productType) - $quantity);
    }

    public function hasEnoughInventory(ProductType $productType, int $quantity): bool
    {
        return $this->getInventoryFor($productType, $quantity) >= $quantity;
    }

    public function getInventoryFor(ProductType $productType): int
    {
        return array_key_exists($productType->name, $this->inventory)
            ? $this->inventory[$productType->name]
            : 0;
    }

    private function updateStore(ProductType $productType, int $newQuantity): void
    {
        $this->inventory[$productType->name] = $newQuantity;
    }

    /*function hasEnoughInventory(ProductType $productType, int $quantity): void;



    function updateStore(ProductType $productType, int $newQuantity): int;*/
}